src/Aspire.Hosting/Extensions/ContainerResourceBuilderExtensions.cs

```diff
@@ -81,6 +81,63 @@ public static IResourceBuilder<ContainerResource> AddContainer(this IDistributed
         builder.Resource.Entrypoint = entrypoint;
         return builder;
     }
+
+    /// <summary>
+    /// Allows overriding the image tag on a container.
+    /// </summary>
+    /// <typeparam name="T">Type of container resource.</typeparam>
+    /// <param name="builder">Builder for the container resource.</param>
+    /// <param name="tag">Tag value.</param>
+    /// <returns></returns>
+    public static IResourceBuilder<T> WithImageTag<T>(this IResourceBuilder<T> builder, string? tag) where T : ContainerResource
```

Comment:
Why is `tag` nullable? It doesn't seem intuitive what `.WithImageTag(null)` means.

Reply:
It allows folks to set it back to whatever the default is. But I agree it looks odd for the API.

Reply:
I don't think that jives well with #2224. We aren't using `latest` anymore as the default. I think this should just be `string tag`. It shouldn't accept `null`.

Reply:
I agree. I'll fix this up.